src/Controller/SelfServiceController.php

```diff
@@ -101,6 +112,18 @@ public function index()
                             $attributeId = $productAttribute->id_product_attribute;
                             break;
                         }
+                        else {
+                            $prefixbarcodewithweight = "27"; //Mario: an welcher globalen Stelle willst du das hinterlegen?
+                            if (strpos($keyword, $prefixbarcodewithweight) === 0){
```

Comment:
die neue methode zur vermeidung von duplicate code kannst du in den CatalogService packen
den instanziierten CatalogService hast du auch im SelfService Controller zur Verfügung.
z.B.
```
public function hasBarcodeWeightPrefix($barcode) { : bool
    return strpos($barcode, BARCODE_WITH_WEIGHT_PREFIX ) === 0);
    //achtung bitte checken ob strpos auch *nur* am anfang des strings sucht, sonst gibts evtl einen parameter dafür oder  du verwendest preg_match mit ^
//    return preg_match('/^'. BARCODE_WITH_WEIGHT_PREFIX  . '/', $string);
}
```